projects/hr-portal/src/app/hr-portal/hr-portal.tsx

```diff
@@ -0,0 +1,145 @@
+  IgrGridToolbarActions,
+  IgrGridToolbarAdvancedFiltering,
+  IgrGridToolbarHiding,
+  IgrGridToolbarPinning,
+  IgrGridToolbarExporter,
+  IgrColumn,
+  IgrPaginator,
+  IgrCellTemplateContext,
+} from 'igniteui-react-grids';
+import { IgrIcon, IgrAvatar, IgrIconButton, IgrButton, registerIcon } from 'igniteui-react';
+import 'igniteui-react-grids/grids/combined';
+import 'igniteui-react-grids/grids/themes/light/fluent.css';
+import { DataService } from './../services/data.service';
+import { icons } from '../data/icons/Icons'
+
+export default function HRPortal() {
+  const [data, setData] = useState<any[]>([]);
+  const [isSorted, setIsSorted] = useState(false);
+  const dataService = React.useMemo(() => new DataService(), []);
+
+  const gridRef = useRef<IgrTreeGrid | null>(null);
+
+  useEffect(() => {
+    dataService.fetchData().then((fetchedData: any) => {
+      setData(fetchedData);
+    });
+
+    icons.forEach((element: { name: string; path: string; category: string }) => {
+      registerIcon(element.name, element.path, element.category);
+    });
+  }, [dataService]);
+
+  const avatarTemplate = (props: { dataContext: IgrCellTemplateContext }) => {
+    const data = props.dataContext.cell.row.data;
+    return (
+      <div className="employeeDiv">
+        <IgrAvatar shape="rounded" src={data.Picture} className="small" />
+        <span>{data.Name}</span>
+      </div>
+    );
+  };
+
+  const countryIconTemplate = (props: { dataContext: IgrCellTemplateContext }) => {
+    const data = props.dataContext.cell.row?.data;
+    if (!data) {
+      return <div>No Data</div>; // Fallback for missing data
+    }
+    return (
+      <div className="flagDiv">
+        <IgrIcon collection="country-icons" name={data.Country} />
+        <span>{data.Location}, {data.Country}</span>
+      </div>
+    );
+  };
+
+  const contactsTemplate = (props: { dataContext: IgrCellTemplateContext }) => {
+    const data = props.dataContext.cell.row.data;
+    return (
+      <div className="center-content small">
+        <a href={`mailto:${data.Email}`}>
+          <IgrIconButton collection="hr-icons" name="mail" variant="flat" />
+        </a>
+        <a href={`tel:${data.Phone}`}>
+          <IgrIconButton collection="hr-icons" name="tel" variant="flat" />
+        </a>
+        <a href="https://www.linkedin.com" target="_blank" rel="noreferrer">
+          <IgrIconButton collection="hr-icons" name="linkedIn" variant="flat" />
+        </a>
+      </div>
+    );
+  };
+
+  const dateTemplate = (props: { dataContext: IgrCellTemplateContext }) => {
+    const data = props.dataContext.cell.row.data;
+    const formattedDate = new Date(data.HireDate).toISOString().split('T')[0];
+    return <>{formattedDate}</>;
+  };
+
+  const clearSorting = useCallback(() => {
+    if (gridRef.current) {
+      gridRef.current.sortingExpressions = [];
+    }
+    setIsSorted(false);
+  }, []);
+
+  const handleSortingChanged = useCallback(() => {
```

Comment:
Why do you need `useCallback` hook here?

Reply:
It is recommended since sorting is passed as a prop to the button and the grid. Using it should avoid unnecessary re-render and improve performance.

Reply:
Have you tested this and did u see a boost in performance? `useCallback` is needed for:
1. If you pass that function as a dependancy to another hook
2. The function is passed as a prop (like you do) to a component that uses memoization. Our components don`t use neither `useMemo` nor `React.memo`. 
I don't see any benefit overcomplicating the code, plus that if you don't need it you could cause a minor performance lost